cdk/cdk_stack.py

```diff
@@ -39,16 +39,24 @@ def __init__(self, scope: cdk.App, id: str, **kwargs) -> None:
 
         apidata_bucket.grant_write(lambda_function)
 
-        api = apigw.LambdaRestApi(
+        second_lambda_function = _lambda.Function(
             self,
-            'WeatherDataAPI',
-            handler=lambda_function,
-            proxy=False
+            'FlattenJsontoCSV',
+            runtime=_lambda.Runtime.PYTHON_3_7,
+            handler='lambda_function.handler',
+            code=_lambda.Code.from_asset('lambda_2'),
```

Comment:
Try to use descriptive names where you can. For example the lambda file name could be "convert_weather-json_to_csv.py". 
Similarly you can specify a function_name="convert-weather-json-to-csv"

Reply:
Yeah sounds good, will keep that in mind from here on out.